### PR TITLE
feat(core): add change username session api

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Add change username session API

## Testing

- [x] UT

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/session/profile.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      body: object({ username: string().regex(usernameRegEx).or(literal('')).nullable() }),
+    }),
+    async (ctx, next) => {
+      const userId = await checkSessionHealth(ctx, provider, verificationTimeout);
+
+      assertThat(userId, new RequestError('auth.unauthorized'));
+
+      const { username } = ctx.guard.body;
+
+      await checkRequiredSignUpIdentifiers({ username });
```
Comment (darcyYe) [inline packages/core/src/routes/session/profile.ts:46]: per the discussion we had before, seems this check on `SignInExp` is unnecessary.

Comment (charIeszhao) [reply packages/core/src/routes/session/profile.ts:46]: No, this is still needed. As we have to re-authenticate the user if the last authentication time is 10 minutes ago

Comment (darcyYe) [reply packages/core/src/routes/session/profile.ts:46]: But the check on authentication time is done in `checkSessionHealth()`?

Comment (charIeszhao) [reply packages/core/src/routes/session/profile.ts:46]: Ah, sorry. My bad. You're right, the authentication time is checked in `checkSessionHealth()`.
And this function is not meant for authentication check, what it does is checking if the required identifier is missing. For example, when user calls this patch API to clear username, the server should be able to throw an error if username is a required sign-up identifier.

However, according to a recent discussion last night, we decided to prohibit the users from removing the username field. So I will remove this check from patch username API and change the body guard accordingly

Comment (charIeszhao) [reply packages/core/src/routes/session/profile.ts]: Code updated and also with some function renamings. PTAL

File: packages/core/src/routes/session/profile.test.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      });
+
+      it('should be able to use null value to clear the field', async () => {
+        const response = await sessionRequest
+          .patch(`${profileRoute}/username`)
+          .send({ username: null });
+
+        expect(response.statusCode).toEqual(200);
+        expect(response.body).toEqual({ ...mockUserResponse, username: null });
+      });
```
Comment (darcyYe) [inline packages/core/src/routes/session/profile.test.ts:91]: This test seems invalid in our use case since `username` can not be deleted.

**Review by darcyYe — APPROVED** [review]
Overall LGTM

**Pull request merged**